pkgs/tools/admin/lexicon/default.nix

```diff
@@ -1,60 +1,27 @@
 { lib
-, python3
 , fetchFromGitHub
+, poetry2nix
+, python3
+, pyproject ? ./pyproject.toml
+, poetrylock ? ./poetry.lock
 }:
 
-with python3.pkgs;
+let
+  pyprojectToml = lib.importTOML pyproject;
+  pyprojectVersion = pyprojectToml.tool.poetry.version;
+in poetry2nix.mkPoetryApplication {
 
-buildPythonApplication rec {
-  pname = "lexicon";
-  version = "3.9.4";
-  format = "pyproject";
+  inherit pyproject poetrylock;
 
-  src = fetchFromGitHub {
+  python = python3;
+
+  projectDir = fetchFromGitHub {
     owner = "AnalogJ";
-    repo = pname;
-    rev = "v${version}";
-    hash = "sha256-TySgIxBEl2RolndAkEN4vCIDKaI48vrh2ocd+CTn7Ow=";
+    repo = "lexicon";
+    rev = "v${pyprojectVersion}";
+    hash = "sha256-z0GaA1O0ctP280QvhdzW7Cxonidz9dOnP6N4RJ+tqfw=";
   };
 
-  nativeBuildInputs = [
-    poetry-core
-  ];
-
-  propagatedBuildInputs = [
-    beautifulsoup4
-    boto3
-    cryptography
-    dnspython
-    future
-    localzone
-    oci
-    pynamecheap
-    pyyaml
-    requests
-    softlayer
-    tldextract
-    transip
-    xmltodict
-    zeep
-  ];
-
-  checkInputs = [
-    mock
-    pytestCheckHook
-    pytest-xdist
-    vcrpy
-  ];
-
-  disabledTestPaths = [
-    # Tests require network access
-    "lexicon/tests/providers/test_auto.py"
-  ];
-
-  pythonImportsCheck = [
-    "lexicon"
-  ];
```

Comment:
Removing testing is also a really bad idea and generally causes a package to rot over time and silently break.